metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -495,8 +496,19 @@ void deleteTopicConfigs(String name) {
         configData.remove(new ConfigResource(Type.TOPIC, name));
     }
 
-    boolean uncleanLeaderElectionEnabledForTopic(String name) {
-        return false; // TODO: support configuring unclean leader election.
+    /**
+     * Check if this topic has "unclean.leader.election.enable" set to true.
+     *
+     * @param topicName            The topic name for the config.
+     * @return true if this topic has uncleanLeaderElection enabled
+     */
+    boolean uncleanLeaderElectionEnabledForTopic(String topicName) {
```

Comment:
While this will work for most cases, there are some cases it won't catch. For example, let's say we create a transaction that has these records:
```
begin_transaction
ConfigRecord(ResourceType=TOPIC, ResourceName="foo", Name="unclean.leader.election", Value="true")
PartitionChangeRecord(... sets leader for foo to -1 and ISR = [2]... )
end_transaction
```

`uncleanLeaderElectionEnabledForTopic` will return false during the transaction since the ConfigRecord has not been applied yet.

There are two ways we could try to deal with this:
1. double down on fixing the edge-triggered method, by passing in a context here that encapsulated pending configuration changes
2. have a periodic sweep that tries to do unclean leader elections on each leaderless partition if possible

For approach number 2, we could make it part of the periodic leader balancing task (which runs every 5 minutes)

Reply:
Good point! Adopt approach 2 to trigger uncleanLeaderElection for no leader partitions when leader rebalancing in this commit: https://github.com/apache/kafka/pull/16284/commits/a9df33af3657099fb33c7c97a31d20e12a9600b5 . Thanks.